db/db_impl.cc

```diff
@@ -565,66 +565,60 @@ void DBImpl::PrintStatistics() {
   }
 }
 
-void DBImpl::MaybeDumpStats() {
-  mutex_.Lock();
-  unsigned int stats_dump_period_sec =
-      mutable_db_options_.stats_dump_period_sec;
-  mutex_.Unlock();
-  if (stats_dump_period_sec == 0) return;
-
-  const uint64_t now_micros = env_->NowMicros();
+void DBImpl::StartTimedTasks() {
+  if (mutable_db_options_.stats_dump_period_sec > 0) {
```

Comment:
`stats_dump_period_sec` is dynamic changeable. How do you handle that?

Reply:
add logic in SetOptions to cancel and relaunch thread